### PR TITLE
Add incident.io integration

### PR DESCRIPTION
**Pull request by rorydbain**
### What does this PR do?

Adds integration for incident.io! 

### Motivation

incident.io has started to show some Datadog monitor links natively, and would like to add an integration tile so we can help our users in setting this up.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

None

### CODE REVIEW

File: incident_io/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# incident.io
+
+![datadog-integration][4]
+
+## Overview
+
+Global technology companies use [incident.io][1] to empower teams to run incidents end-to-end. Rapidly fix and learn from incidents, so you can build more resilient products.
```
Comment (urseberry) [inline incident_io/README.md:7]:
```suggestion
Global technology companies use [incident.io][1] to empower teams to run incidents end-to-end. Fix and learn from incidents, so you can build more resilient products.
```

File: incident_io/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+When you trigger incidents using PagerDuty or OpsGenie, incident.io will automatically pull through any associated information about your Datadog monitors, putting all the information you need right in your incident channel.
```
Comment (urseberry) [inline incident_io/README.md:9]:
```suggestion
When you trigger an incident using PagerDuty or OpsGenie, incident.io automatically pulls through any associated information about your Datadog monitors, putting all the information you need in your incident channel.
```

File: incident_io/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+## Setup
+
+To get started, you need to first connect PagerDuty or OpsGenie to Datadog, then setup incident triggers inside incident.io - details on how to do that can be found in the [incident.io documentation][2]
```
Comment (urseberry) [inline incident_io/README.md:13]:
```suggestion
To get started, connect PagerDuty or OpsGenie to Datadog. Then, set up incident triggers inside incident.io. See the [incident.io Datadog integration][2] for details.
```

File: incident_io/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+## Support
+
+Contact incident.io's support team via Intercom on [our website][1], or reach out to our [support team][3].
```
Comment (urseberry) [inline incident_io/README.md:17]:
```suggestion
Contact incident.io's support team via Intercom on [the website][1], or reach out to the [support team][3].
```

**Review by urseberry — APPROVED** [review]

**Comment by eho1307** [comment]
Hi @rorydbain, Thank you for opening this PR! The Datadog Marketplace team has tried reaching out to your colleague, Isaac, over email but haven't heard back. We typically require integrations to send data into Datadog rather than just pull it out. Could you update your PR to be bi-directional? Happy to jump on a call to further explain.

**Comment by rorydbain** [comment]
> Hi @rorydbain, Thank you for opening this PR! The Datadog Marketplace team has tried reaching out to your colleague, Isaac, over email but haven't heard back. If you're planning on also listing on our Marketplace, e typically require integrations to send data into Datadog rather than just pull it out. Happy to jump on a call to further explain. If you're planning on just listing a free integration, that works as well. Just let us know!

Hey! Thanks for the info. Our initial integration is just ingesting data, but we very well may push events in the future. I'll reopen this later, thanks!

**Pull request closed without merging**